File: backend/library/models/book.py
from pydantic import BaseModel
from typing import Optional

class Book(BaseModel):
    code: Optional[str] = None
    title: Optional[str] = None
    description: Optional[str] = None
    author: Optional[str] = None
    quantity: Optional[int] = None
    reserved_quantity: Optional[int] = 0

class UpdateBook(BaseModel):
    title: Optional[str] = None
    description: Optional[str] = None
    quantity: Optional[int] = None
    reserved_quantity: Optional[int] = 0

File: frontend/lib/features/home/domain/entities/book.dart
import 'package:equatable/equatable.dart';

class Book extends Equatable {
  final String code;
  final String title;
  final String author;
  final String description;
  final int quantity;
  final int reservedQuantity;

  const Book({
    required this.code,
    required this.title,
    required this.author,
    required this.description,
    required this.quantity,
    required this.reservedQuantity,
  });

  @override
  List<Object?> get props => [
        code,
        title,
        author,
        description,
        quantity,
        reservedQuantity,
      ];
}
